pkg/conference/data_channel_message_processor.go

```diff
@@ -28,47 +36,61 @@ func (c *Conference) processSelectDCMessage(participant *Participant, msg event.
 		}
 	}
 
-	// Subscribe to the found tracks.
+	// Subscribe to the found tracks
 	for _, track := range tracks {
+		participant.logger.WithField("track_id", track.ID()).Debug("Subscribing to track")
 		if err := participant.peer.SubscribeTo(track); err != nil {
 			participant.logger.Errorf("Failed to subscribe to track: %v", err)
 			return
 		}
 	}
-}
-
-func (c *Conference) processAnswerDCMessage(participant *Participant, msg event.SFUMessage) {
-	participant.logger.Info("Received SDP answer over DC")
 
-	if err := participant.peer.ProcessSDPAnswer(msg.SDP); err != nil {
-		participant.logger.Errorf("Failed to set SDP answer: %v", err)
-		return
-	}
+	// TODO: Handle unsubscribe
 }
 
-func (c *Conference) processPublishDCMessage(participant *Participant, msg event.SFUMessage) {
-	participant.logger.Info("Received SDP offer over DC")
+func (c *Conference) processNegotiateDCMessage(participant *Participant, msg event.FocusCallNegotiateEventContent) {
+	participant.streamMetadata = msg.SDPStreamMetadata
 
-	answer, err := participant.peer.ProcessSDPOffer(msg.SDP)
-	if err != nil {
-		participant.logger.Errorf("Failed to set SDP offer: %v", err)
-		return
-	}
+	if msg.Description.Type == event.CallDataTypeOffer {
```

Comment:
We have `if`, `else if`, and `else` for checking the type of the event here. It sounds like a good candidate for a `switch` statement to me. It feels like it would be a bit more elegant to switch for the type (good use case for a `switch`).